ratis-common/src/main/java/org/apache/ratis/protocol/RaftPeer.java

```diff
@@ -141,11 +181,21 @@ public RaftPeerId getId() {
     return id;
   }
 
-  /** @return The RPC address of the peer. */
+  /** @return The RPC address of the peer for server-server communication. */
   public String getAddress() {
     return address;
   }
 
+  /** @return The RPC address of the peer for admin operations. */
+  public String getAdminAddress() {
+    return adminAddress != null ? adminAddress : address;
```

Comment:
getAdminAddress() should just return adminAddress.  The callers should fallback to use getAddress() when getAdminAddress() returns null.  

<img width="932" alt="image" src="https://user-images.githubusercontent.com/907380/105442392-c9810580-5ca4-11eb-8c1b-0da0d6dfb043.png">

In the current usages, we already use Optional.ofNullable(..) to handle the case getAdminAddress() returning null.

Reply:
Good idea, done.  Thanks.